From b84e17579c2f8de1487113f5bcf528429e571264 Mon Sep 17 00:00:00 2001
From: Ruslan Bay <ruslanbey@proton.me>
Date: Thu, 26 Feb 2026 17:21:26 +0100
Subject: [PATCH 32/54] media: intel/ipu4: Use v4l2_async_connection for async
 sensor discovery

Replace v4l2_async_subdev with the newer v4l2_async_connection type. This reflects
the evolution of the V4L2 async framework to better represent connections between
sensor and ISP components.
---
 drivers/media/pci/intel/ipu-isys.c | 2 +-
 drivers/media/pci/intel/ipu-isys.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-isys.c b/drivers/media/pci/intel/ipu-isys.c
index b3af99469793..e4e7cd97da20 100644
--- a/drivers/media/pci/intel/ipu-isys.c
+++ b/drivers/media/pci/intel/ipu-isys.c
@@ -398,7 +398,7 @@ static struct media_device_ops isys_mdev_ops = {
 /* The .bound() notifier callback when a match is found */
 static int isys_notifier_bound(struct v4l2_async_notifier *notifier,
 			       struct v4l2_subdev *sd,
-			       struct v4l2_async_subdev *asd)
+			       struct v4l2_async_connection *asd)
 {
 	struct ipu_isys *isys =
 		container_of(notifier, struct ipu_isys, notifier);
diff --git a/drivers/media/pci/intel/ipu-isys.h b/drivers/media/pci/intel/ipu-isys.h
index 606c0ebfc96a..3511faca318d 100644
--- a/drivers/media/pci/intel/ipu-isys.h
+++ b/drivers/media/pci/intel/ipu-isys.h
@@ -65,7 +65,7 @@ struct ipu_isys_csi2_config {
 };
 
 struct sensor_async_sd {
-	struct v4l2_async_subdev asd;
+	struct v4l2_async_connection asd;
 	struct ipu_isys_csi2_config csi2;
 };
 
-- 
2.51.0

